core/src/main/java/com/graphhopper/routing/weighting/custom/ExpressionVisitor.java

```diff
@@ -0,0 +1,223 @@
+                        StringEncodedValue ev = lookup.getStringEncodedValue(lhVarAsString);
+                        String str = ((Java.StringLiteral) binOp.rhs).value;
+                        int integ = ev.indexOf(str.substring(1, str.length() - 1));
+                        if (integ == 0) integ = -1; // 0 means not found and this should always trigger inequality
+                        replacements.put(startRH, new Replacement(startRH, str.length(), "" + integ));
+                    }
+                } else if (binOp.rhs instanceof Java.AmbiguousName && ((Java.AmbiguousName) binOp.rhs).identifiers.length == 1) {
+                    // Make enum explicit as NO or OTHER can occur in other enums so convert "toll == NO" to "toll == Toll.NO"
+                    String rhValue = ((Java.AmbiguousName) binOp.rhs).identifiers[0];
+                    if (nameValidator.isValid(lhVarAsString) && rhValue.toUpperCase(Locale.ROOT).equals(rhValue)) {
+                        if (!eqOps)
+                            throw new IllegalArgumentException("Operator " + binOp.operator + " not allowed for enum");
+                        String value = toEncodedValueClassName(binOp.lhs.toString());
+                        replacements.put(startRH, new Replacement(startRH, rhValue.length(), value + "." + rhValue));
+                    }
+                }
+            }
+            return binOp.lhs.accept(this) && binOp.rhs.accept(this);
+        }
+        return false;
+    }
+
+    @Override
+    public Boolean visitPackage(Java.Package p) {
+        return false;
+    }
+
+    @Override
+    public Boolean visitType(Java.Type t) {
+        return false;
+    }
+
+    @Override
+    public Boolean visitConstructorInvocation(Java.ConstructorInvocation ci) {
+        return false;
+    }
+
+    static void parseExpressions(StringBuilder expressions, NameValidator nameInConditionValidator, String exceptionInfo,
+                                 Set<String> createObjects, List<Statement> list, EncodedValueLookup lookup, String lastStmt) {
+
+        for (Statement statement : list) {
+            if (statement.getKeyword() == Statement.Keyword.ELSE) {
+                if (!Helper.isEmpty(statement.getExpression()))
+                    throw new IllegalArgumentException("expression must be empty but was " + statement.getExpression());
+
+                expressions.append("else {" + statement.getOperation().build(statement.getValue()) + "; }\n");
+            } else if (statement.getKeyword() == Statement.Keyword.ELSEIF || statement.getKeyword() == Statement.Keyword.IF) {
+                ExpressionVisitor.ParseResult parseResult = parseExpression(statement.getExpression(), nameInConditionValidator, lookup);
+                if (!parseResult.ok)
+                    throw new IllegalArgumentException(exceptionInfo + " invalid simple condition: " + statement.getExpression());
```

Comment:
I got this error when I used the 'Example' of the flex box in GH maps because I was missing the `max_weight` encoded value. The error read: 

```
Cannot compile expression , priority_user_statements invalid simple condition: max_weight > 3
```

Can we provide a more meaningful error message, e.g. state that the `max_weight` encoded value is missing, and maybe which ones are available? Debugging custom models is probably very frustrating without detailed error messages.

Reply:
Can you check if the improved error messages would have been helpful in your case?

https://github.com/graphhopper/graphhopper/pull/2209/commits/d6552b24d0b592ceecee8fdd18f87522c1a8a17c#diff-dafd159caf8461d50b0bda954bc5d78540d453751585e81682e22cc3ae09bce9R132

Reply:
Now I get

```
Cannot compile expression , priority_user_statements invalid expression "max_weight < 3": encoded value 'max_weight' not available
```

Yes that's better. The first part still looks a bit cryptic, but at least now we are told `max_weight` is missing.

Reply:
Ok. Now it should be:

```
Cannot compile expression in 'priority' entry, invalid expression "max_weight < 3": encoded value 'max_weight' not available
```